CYF-ITP-South Africa | Simphiwe Mabuya | Module-Structuring-and-Testing-Data : Sprint 1 | week 1

Simphiwe-Mabuya opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

cjyuan commented on Sprint-1/exercises/count.js:
```diff
@@ -2,5 +2,9 @@ let count = 0;
 
 count = count + 1;
 
+// on line three 1 is added to the count variable by using the assignment operator =
```
We have a term to describe operation like `count = count + 1` in programming. Would you try feeding the code to ChatGPT to see how else the code can be described? You may learn some terminology or more concise way to describe code from time to time.

cjyuan commented on Sprint-1/exercises/paths.js:
```diff
@@ -14,5 +14,11 @@ const lastSlashIndex = filePath.lastIndexOf("/");
+
+
 // Create a variable to store the dir part of the filePath variable
+const filePathDir = filePath.slice(- 41);
```
Your code should work for any valid file path, not just the sample value used in this exercise.

Can you rewrite your code so that it will work for any file path? For example,
`/path1/path2.2/index.html`

cjyuan commented on Sprint-1/exercises/random.js:
```diff
@@ -2,6 +2,7 @@ const minimum = 1;
 const maximum = 100;
 
 const num = Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;
+console.log(num); // num represents a random number generated between 1 and 100 (minimum & maximum) : a random number is generated using Math.random() then brought into a whole integer using Math.floor()
```
To test your understanding, how would you write an expression (without using any variable in the expression) that yields a random integer between -5 and 3 (including -5 and 3)?

cjyuan commented on Sprint-1/explore/chrome.md:
```diff
@@ -11,8 +11,11 @@ In the Chrome console,
 What effect does calling the `prompt` function have?
+- The function pops up a prompt box for the user to enter an input 
 What is the return value of `prompt`?
+- The return value is whatever the user inputs into the prompt box eg a name in this case
```
How can you tell if the user clicked "OK" or "Cancel"?

cjyuan commented on Sprint-1/explore/objects.md:
```diff
@@ -5,12 +5,17 @@ In this activity, we'll explore some additional concepts that you'll encounter i
 Answer the following questions:
 
 What does `console` store?
+- Console stores values/variables that users input / pass to the console 
```
The description is not quite correct. Would you ask ChatGPT for a second opinion, or to lookup MDN reference to get a more accurate description?

cjyuan commented on Sprint-1/interpret/time-format.js:
```diff
@@ -13,12 +13,24 @@ console.log(result);
+
 // c) Using documentation, explain what the expression movieLength % 60 represents
 
+// This expression represents a calculation that will return a remainder of movieLength seconds after it's divided by 60. The (%) modulus operator is used to find the remainder after dividing one number by another.  
```
This is a literal translation of the code; it does not give any insight what the code is trying to calculate. Can you describe what the code is trying to calculate, in lament terms?

cjyuan commented on Sprint-1/interpret/to-pounds.js:
```diff
@@ -24,4 +22,9 @@ console.log(`£${pounds}.${pence}`);
+// 2. const penceStringWithoutTrailingP is assigned the value of penceString without the trailing p. Calling substring() on penceString and appending the .length method - 1 removes the p from 399p.
+// 3. const paddedPenceNumberString is assigned a padded a value of penceStringWithoutTrailingP where the function padStart has been called to generate a string of a specified length, in this case the length is 3. 
+// 4. const pounds on this line the last two digits are removed from paddedPenceNumberString by calling the substring() function to extract characters(none in this case for 0 was passed as an argument, making sure nothing else is removed) and appending the .length method - 2 (removing 99).
+// 5. const pence is assigned the new value paddedPenceNumberString which is now 99 due to the .length method appended on paddedPenceNumberString where substring and padEnd functions were also called so to reach a specified length.  
```
Can we expect the code to work as intended even if we deleted `.padEnd(2, "0")` from the code?

cjyuan reviewed (commented):
Code is generally good. Just a few bugs.

Simphiwe-Mabuya commented:
Morning CJ

I am sure that this will find you well.
Thanks a million for the review and constructive feedback. I revert back to
you as soon as I am done making the changes/implementing your suggestions.

Thanks a million yet again for taking your time to review my PR.

Best regards
Simphiwe Mabuya

On Wed, 13 Nov 2024 at 12:35, CJ Yuan ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> Code is generally good. Just a few bugs.
> ------------------------------
>
> In Sprint-1/errors/3.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839906323>
> :
>
> > @@ -1,9 +1,12 @@
> -const cardNumber = 4533787178994213;
> +const cardNumber = [4,5,3,3,7,8,7,1,7,8,9,9,4,2,1,3];
>
> If you were not allowed to modify the original value of cardNumber (i.e.,
> keep it as a number type value). how would you rewrite your code to extract
> the last 4 digits?
> ------------------------------
>
> In Sprint-1/exercises/count.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839909498>
> :
>
> > @@ -2,5 +2,9 @@ let count = 0;
>
>  count = count + 1;
>
> +// on line three 1 is added to the count variable by using the assignment operator =
>
> We have a term to describe operation like count = count + 1 in
> programming. Would you try feeding the code to ChatGPT to see how else the
> code can be described? You may learn some terminology or more concise way
> to describe code from time to time.
> ------------------------------
>
> In Sprint-1/exercises/paths.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839914217>
> :
>
> >  // Create a variable to store the dir part of the filePath variable
> +const filePathDir = filePath.slice(- 41);
>
> Your code should work for any valid file path, not just the sample value
> used in this exercise.
>
> Can you rewrite your code so that it will work for any file path? For
> example,
> /path1/path2.2/index.html
> ------------------------------
>
> In Sprint-1/exercises/random.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839916467>
> :
>
> > @@ -2,6 +2,7 @@ const minimum = 1;
>  const maximum = 100;
>
>  const num = Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;
> +console.log(num); // num represents a random number generated between 1 and 100 (minimum & maximum) : a random number is generated using Math.random() then brought into a whole integer using Math.floor()
>
> To test your understanding, how would you write an expression (without
> using any variable in the expression) that yields a random integer between
> -5 and 3 (including -5 and 3)?
> ------------------------------
>
> In Sprint-1/explore/chrome.md
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839918207>
> :
>
> >  What is the return value of `prompt`?
> +- The return value is whatever the user inputs into the prompt box eg a name in this case
>
> How can you tell if the user clicked "OK" or "Cancel"?
> ------------------------------
>
> In Sprint-1/explore/objects.md
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839926651>
> :
>
> >
>  Answer the following questions:
>
>  What does `console` store?
> +- Console stores values/variables that users input / pass to the console
>
> The description is not quite correct. Would you ask ChatGPT for a second
> opinion, or to lookup MDN reference to get a more accurate description?
> ------------------------------
>
> In Sprint-1/interpret/time-format.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839929920>
> :
>
> >  // c) Using documentation, explain what the expression movieLength % 60 represents
>
> +// This expression represents a calculation that will return a remainder of movieLength seconds after it's divided by 60. The (%) modulus operator is used to find the remainder after dividing one number by another.
>
> This is a literal translation of the code; it does not give any insight
> what the code is trying to calculate. Can you describe what the code is
> trying to calculate, in lament terms?
> ------------------------------
>
> In Sprint-1/interpret/to-pounds.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839933229>
> :
>
> > @@ -24,4 +22,9 @@ console.log(`£${pounds}.${pence}`);
>  // Try and describe the purpose / rationale behind each step
>
>  // To begin, we can start with
> -// 1. const penceString = "399p": initialises a string variable with the value "399p"
> +// 1. const penceString = "399p": initializes a string variable with the value "399p"
> +// 2. const penceStringWithoutTrailingP is assigned the value of penceString without the trailing p. Calling substring() on penceString and appending the .length method - 1 removes the p from 399p.
> +// 3. const paddedPenceNumberString is assigned a padded a value of penceStringWithoutTrailingP where the function padStart has been called to generate a string of a specified length, in this case the length is 3.
> +// 4. const pounds on this line the last two digits are removed from paddedPenceNumberString by calling the substring() function to extract characters(none in this case for 0 was passed as an argument, making sure nothing else is removed) and appending the .length method - 2 (removing 99).
> +// 5. const pence is assigned the new value paddedPenceNumberString which is now 99 due to the .length method appended on paddedPenceNumberString where substring and padEnd functions were also called so to reach a specified length.
>
> Can we expect the code to work as intended even if we deleted .padEnd(2,
> "0") from the code?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#pullrequestreview-2432550611>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BIXBQLIMJTKM355RFUWJTQL2AMTPTAVCNFSM6AAAAABRURWXGOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMZSGU2TANRRGE>
> .
> You are receiving this because you authored the thread.Message ID:
> <CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132/review/2432550611
> @github.com>
>

Simphiwe-Mabuya commented:
Greetings again CJ, I trust this will find you in great spirits.

1 Sprint-1/errors/3.js
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839906323>:


I've managed to rewrite my code so it can extract the last four characters
without modifying the original value of cardNumber

2 Sprint-1/exercises/count.js
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839909498>:


Here I've given the correct term to describe an operation like count =
count + 1 in programming

3 Sprint-1/exercises/paths.js
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839914217>:


I believe I've also managed to rewrite my code so that it will work for any
file path.

4 Sprint-1/exercises/random.js
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839916467>:


To broaden my understanding regarding Math.random() I rewrote an expression
without using any variable in the expression and was able to yield a random
integer between -5 and 3 (including -5 and 3).

5 Sprint-1/explore/chrome.md
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839918207>
:

I figured out and explained what happens if the user doesn't click OK or
Cancel.

6 Sprint-1/explore/objects.md
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839926651>
:

I also managed to find a more correct way to describe what console stores.

7 Sprint-1/interpret/time-format.js
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839929920>
:

I gave a much detailed and clear description of what the code is trying to
calculate.

8 Sprint-1/interpret/to-pounds.js
<https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839933229>
:

I found out if the code will continue working if I were to remove padEnd()
and explained what happens in depth.

Thanks again Mr CJ for your constructive review and your assistance
regarding pushing me to figure things out by myself and how helpful it is
to utilise ChatGpT and MDN or W3School documentation. Please do enjoy the
rest of your day.

Looking forward to hearing more of your thoughts in regards to my reply to
your feedback.

Best regards
Simphiwe Mabuya


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

On Wed, 13 Nov 2024 at 12:35, CJ Yuan ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> Code is generally good. Just a few bugs.
> ------------------------------
>
> In Sprint-1/errors/3.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839906323>
> :
>
> > @@ -1,9 +1,12 @@
> -const cardNumber = 4533787178994213;
> +const cardNumber = [4,5,3,3,7,8,7,1,7,8,9,9,4,2,1,3];
>
> If you were not allowed to modify the original value of cardNumber (i.e.,
> keep it as a number type value). how would you rewrite your code to extract
> the last 4 digits?
> ------------------------------
>
> In Sprint-1/exercises/count.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839909498>
> :
>
> > @@ -2,5 +2,9 @@ let count = 0;
>
>  count = count + 1;
>
> +// on line three 1 is added to the count variable by using the assignment operator =
>
> We have a term to describe operation like count = count + 1 in
> programming. Would you try feeding the code to ChatGPT to see how else the
> code can be described? You may learn some terminology or more concise way
> to describe code from time to time.
> ------------------------------
>
> In Sprint-1/exercises/paths.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839914217>
> :
>
> >  // Create a variable to store the dir part of the filePath variable
> +const filePathDir = filePath.slice(- 41);
>
> Your code should work for any valid file path, not just the sample value
> used in this exercise.
>
> Can you rewrite your code so that it will work for any file path? For
> example,
> /path1/path2.2/index.html
> ------------------------------
>
> In Sprint-1/exercises/random.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839916467>
> :
>
> > @@ -2,6 +2,7 @@ const minimum = 1;
>  const maximum = 100;
>
>  const num = Math.floor(Math.random() * (maximum - minimum + 1)) + minimum;
> +console.log(num); // num represents a random number generated between 1 and 100 (minimum & maximum) : a random number is generated using Math.random() then brought into a whole integer using Math.floor()
>
> To test your understanding, how would you write an expression (without
> using any variable in the expression) that yields a random integer between
> -5 and 3 (including -5 and 3)?
> ------------------------------
>
> In Sprint-1/explore/chrome.md
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839918207>
> :
>
> >  What is the return value of `prompt`?
> +- The return value is whatever the user inputs into the prompt box eg a name in this case
>
> How can you tell if the user clicked "OK" or "Cancel"?
> ------------------------------
>
> In Sprint-1/explore/objects.md
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839926651>
> :
>
> >
>  Answer the following questions:
>
>  What does `console` store?
> +- Console stores values/variables that users input / pass to the console
>
> The description is not quite correct. Would you ask ChatGPT for a second
> opinion, or to lookup MDN reference to get a more accurate description?
> ------------------------------
>
> In Sprint-1/interpret/time-format.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839929920>
> :
>
> >  // c) Using documentation, explain what the expression movieLength % 60 represents
>
> +// This expression represents a calculation that will return a remainder of movieLength seconds after it's divided by 60. The (%) modulus operator is used to find the remainder after dividing one number by another.
>
> This is a literal translation of the code; it does not give any insight
> what the code is trying to calculate. Can you describe what the code is
> trying to calculate, in lament terms?
> ------------------------------
>
> In Sprint-1/interpret/to-pounds.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1839933229>
> :
>
> > @@ -24,4 +22,9 @@ console.log(`£${pounds}.${pence}`);
>  // Try and describe the purpose / rationale behind each step
>
>  // To begin, we can start with
> -// 1. const penceString = "399p": initialises a string variable with the value "399p"
> +// 1. const penceString = "399p": initializes a string variable with the value "399p"
> +// 2. const penceStringWithoutTrailingP is assigned the value of penceString without the trailing p. Calling substring() on penceString and appending the .length method - 1 removes the p from 399p.
> +// 3. const paddedPenceNumberString is assigned a padded a value of penceStringWithoutTrailingP where the function padStart has been called to generate a string of a specified length, in this case the length is 3.
> +// 4. const pounds on this line the last two digits are removed from paddedPenceNumberString by calling the substring() function to extract characters(none in this case for 0 was passed as an argument, making sure nothing else is removed) and appending the .length method - 2 (removing 99).
> +// 5. const pence is assigned the new value paddedPenceNumberString which is now 99 due to the .length method appended on paddedPenceNumberString where substring and padEnd functions were also called so to reach a specified length.
>
> Can we expect the code to work as intended even if we deleted .padEnd(2,
> "0") from the code?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#pullrequestreview-2432550611>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BIXBQLIMJTKM355RFUWJTQL2AMTPTAVCNFSM6AAAAABRURWXGOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMZSGU2TANRRGE>
> .
> You are receiving this because you authored the thread.Message ID:
> <CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132/review/2432550611
> @github.com>
>

cjyuan commented on Sprint-1/interpret/to-pounds.js:
```diff
@@ -24,4 +22,11 @@ console.log(`£${pounds}.${pence}`);
+// 5. const pence is assigned the new value paddedPenceNumberString which is now 99 due to the .length method appended on paddedPenceNumberString where substring and padEnd functions were also called so to reach a specified length.  
+// 6. Lastly the console.log function is called using interpolation (template literals) to concatenate the string variables with the pound sign and a dot.
+
+// 7. The code still works though I've removed padEnd(). However, padEnd() is used to make sure that the resulting string is always at least two characters long. It is then necessary or helpful if you want to ensure you always have exactly two digits for the pence part. If we remove padEnd(), then pence might not be exactly two characters in some cases.
```
Can you come up with any case where pence will not have two characters if we removed `pandEnd(2, "0")`? I challenge you. 😄

cjyuan reviewed (commented):
Code look good. I left one puzzle for your to solve. In any case you can mark this PR as completed anytime.

Simphiwe-Mabuya commented:
Thanks a million yet again CJ for your assistance and guidance with this
Sprint, it really means a lot to me. 🙏
Also thanks for the challenge 😄😁 let me try it out ...

I''ll go ahead and mark the PR as completed 🙏

Looking forward to more of your feedback when you get time to check my
other tickets.
Have a great day ahead.

Best regards
Simphiwe Mabuya

On Wed, 20 Nov 2024 at 04:18, CJ Yuan ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> Code look good. I left one puzzle for your to solve. In any case you can
> mark this PR as completed anytime.
> ------------------------------
>
> In Sprint-1/interpret/to-pounds.js
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#discussion_r1849406788>
> :
>
> > @@ -24,4 +22,11 @@ console.log(`£${pounds}.${pence}`);
>  // Try and describe the purpose / rationale behind each step
>
>  // To begin, we can start with
> -// 1. const penceString = "399p": initialises a string variable with the value "399p"
> +// 1. const penceString = "399p": initializes a string variable with the value "399p"
> +// 2. const penceStringWithoutTrailingP is assigned the value of penceString without the trailing p. Calling substring() on penceString and appending the .length method - 1 removes the p from 399p.
> +// 3. const paddedPenceNumberString is assigned a padded value of penceStringWithoutTrailingP where the function padStart has been called to generate a string of a specified length, in this case the length is 3.
> +// 4. const pounds: on this line the last two digits are removed from paddedPenceNumberString by calling the substring() function to extract characters(none in this case for 0 was passed as an argument, making sure nothing else is removed) and appending the .length method - 2 (removing 99).
> +// 5. const pence is assigned the new value paddedPenceNumberString which is now 99 due to the .length method appended on paddedPenceNumberString where substring and padEnd functions were also called so to reach a specified length.
> +// 6. Lastly the console.log function is called using interpolation (template literals) to concatenate the string variables with the pound sign and a dot.
> +
> +// 7. The code still works though I've removed padEnd(). However, padEnd() is used to make sure that the resulting string is always at least two characters long. It is then necessary or helpful if you want to ensure you always have exactly two digits for the pence part. If we remove padEnd(), then pence might not be exactly two characters in some cases.
>
> Can you come up with any case where pence will not have two characters if
> we removed pandEnd(2, "0")? I challenge you. 😄
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132#pullrequestreview-2447200726>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BIXBQLPTUUHWGHVQQ63P4ML2BPWPPAVCNFSM6AAAAABRURWXGOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDINBXGIYDANZSGY>
> .
> You are receiving this because you authored the thread.Message ID:
> <CodeYourFuture/Module-Structuring-and-Testing-Data/pull/132/review/2447200726
> @github.com>
>
